packages/valory/connections/test_libp2p/tests/conftest.py

```diff
@@ -0,0 +1,217 @@
+    except (ImageNotFound, NotFound) as e:
+        logger.error(e)
+    finally:
+        container.remove()
+
+
+def launch_many_containers(
+    image: DockerImage, timeout: float = 2.0, max_attempts: int = 10
+) -> Generator[DockerImage, None, None]:
+    """Launch many containers from an image."""
+    _pre_launch(image)
+    containers = image.create()
+    for container in containers:
+        _start_container(image, container, timeout, max_attempts)
+    yield image
+    for container in containers:
+        _stop_container(container, image.tag)
+
+
+LOCAL_ADDRESS = "0.0.0.0"  # nosec
+ACN_CONFIGURATION: Dict[str, str] = dict(
+    AEA_P2P_ID="54562eb807d2f80df8151db0a394cac72e16435a5f64275c277cae70308e8b24",
+    AEA_P2P_URI_PUBLIC=f"{LOCAL_ADDRESS}:5000",
+    AEA_P2P_URI=f"{LOCAL_ADDRESS}:5000",
+    AEA_P2P_DELEGATE_URI=f"{LOCAL_ADDRESS}:11000",
+    AEA_P2P_URI_MONITORING=f"{LOCAL_ADDRESS}:8080",
+    ACN_LOG_FILE="/acn/libp2p_node.log",
+)
+
+
+@pytest.fixture(scope="session")
+def acn_configuration():
+    """Get the ACN Node configuration for testing purposes."""
+    return ACN_CONFIGURATION
+
+
+@contextmanager
+def _acn_context(
+    acn_configuration: Dict,
+    timeout: float = 2.0,
+    max_attempts: int = 10,
+):
+    client = docker.from_env()
+    image = ACNNodeDockerImage(client, config=acn_configuration)
+    yield from _launch_image(image, timeout=timeout, max_attempts=max_attempts)
+
+
+@pytest.mark.integration
+@pytest.mark.ledger
+@pytest.fixture(scope="class")
+def acn_node(
+    acn_configuration,
+    timeout: float = 2.0,
+    max_attempts: int = 10,
+):
+    """Launch the ACN image."""
+    with _acn_context(acn_configuration, timeout, max_attempts) as image:
+        yield image
+
+
+@pytest.mark.integration
+class UseACNNode:
+    """Inherit from this class to an ACN Node."""
+
+    @pytest.fixture(autouse=True)
+    def _start_acn_node(self, acn_node):
+        """Start a ACN Node image."""
+
+
+@contextmanager
+def _acn_multiple_nodes_context(
+    acn_configuration: Dict,
+    timeout: float = 2.0,
+    max_attempts: int = 10,
+):
+    client = docker.from_env()
+    image = ACNWithBootstrappedEntryNodesDockerImage(client, config=acn_configuration)
+    yield from launch_many_containers(image, timeout, max_attempts)
+
+
+@pytest.mark.ledger
+@pytest.mark.integration
+class ACNWithBootstrappedEntryNodes:
+    """Inherit from this class to an ACN Node."""
+
+    start_local: bool = False
+
+    @pytest.fixture(autouse=True, scope="class")
+    def _start_acn(
+        self, acn_configuration, timeout: float = 2.0, max_attempts: int = 10
+    ):
+        """Start a series of ACN Node images."""
+        if self.start_local:
+            logging.info("Starting up local ACN nodes")
+            with _acn_multiple_nodes_context(
+                acn_configuration, timeout, max_attempts
+            ) as image:
+                yield image
+        else:
+            yield
```

Comment:
again not great. I tried several things, maybe factory pattern is preferred here but not sure. Other ways to make use of the local ACN configurable might be using `@pytest.mark` and `request`, but that means one need to change the marker on the test just the same as setting class attribute now...